homeassistant/components/bayesian/config_flow.py

```diff
@@ -0,0 +1,579 @@
+        {
+            vol.Required(CONF_INDEX): vol.In(
+                {
+                    str(index): f"{config.get(CONF_NAME)} ({config[CONF_PLATFORM]})"
+                    for index, config in enumerate(handler.options[CONF_OBSERVATIONS])
+                },
+            )
+        }
+    )
+
+
+async def _get_remove_observation_schema(
+    handler: SchemaCommonFlowHandler,
+) -> vol.Schema:
+    """Return schema for observation removal."""
+    return vol.Schema(
+        {
+            vol.Required(CONF_INDEX): cv.multi_select(
+                {
+                    str(index): f"{config.get(CONF_NAME)} ({config[CONF_PLATFORM]})"
+                    for index, config in enumerate(handler.options[CONF_OBSERVATIONS])
+                },
+            )
+        }
+    )
+
+
+async def _get_edit_observation_schema(
+    handler: SchemaCommonFlowHandler,
+) -> vol.Schema:
+    """Select which schema to return depending on which observation type it is."""
+
+    observations: list[dict[str, Any]] = handler.options[CONF_OBSERVATIONS]
+    selected_idx = int(handler.options[CONF_INDEX])
+
+    if observations[selected_idx][CONF_PLATFORM] == str(ObservationTypes.STATE):
+        return STATE_SUBSCHEMA
+    if observations[selected_idx][CONF_PLATFORM] == str(ObservationTypes.NUMERIC_STATE):
+        return NUMERIC_STATE_SUBSCHEMA
+
+    return TEMPLATE_SUBSCHEMA
+
+
+async def _choose_observation_step(
+    user_input: dict[str, Any],
+) -> ConfigFlowSteps | None:
+    """Return next step_id for options flow according to template_type."""
+    if user_input.get("add_another", False):
+        return ConfigFlowSteps.OBSERVATION_SELECTOR
+    return None
+
+
+async def _get_base_suggested_values(
+    handler: SchemaCommonFlowHandler,
+) -> dict[str, Any]:
+    """Return suggested values for the sensor base options."""
+
+    return _convert_fractions_to_percentages(dict(handler.options))
+
+
+async def _get_edit_observation_suggested_values(
+    handler: SchemaCommonFlowHandler,
+) -> dict[str, Any]:
+    """Return suggested values for observation editing."""
+
+    idx = int(handler.options[CONF_INDEX])
+    return _convert_fractions_to_percentages(
+        dict(handler.options[CONF_OBSERVATIONS][idx])
+    )
+
+
+async def _validate_user(
+    handler: SchemaCommonFlowHandler, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate the threshold mode, and set limits to None if not set."""
+    if user_input[CONF_PRIOR] == 0:
+        raise SchemaFlowError("prior_low_error")
+    if user_input[CONF_PRIOR] == 100:
+        raise SchemaFlowError("prior_high_error")
+    if user_input[CONF_PROBABILITY_THRESHOLD] == 0:
+        raise SchemaFlowError("threshold_low_error")
+    if user_input[CONF_PROBABILITY_THRESHOLD] == 100:
+        raise SchemaFlowError("threshold_high_error")
+    user_input = _convert_percentages_to_fractions(user_input)
+    return {**user_input}
+
+
+async def _validate_observation_setup(
```

Comment:
It would be good to call some additional, platform-specific checks, i.e check the `to_state` is acutally a valid possible state, and to call some of the built in numeric_state validations in https://github.com/home-assistant/home-assistant.io/pull/33186 once it is merged.